src/borg/archiver.py

```diff
@@ -587,6 +590,8 @@ def peek_and_store_hardlink_masters(item, matched):
         if pi:
             # clear progress output
             pi.finish()
+        if warning:
+            return EXIT_WARNING
         return self.exit_code
```

Comment:
hmm, looks like there are 3 ways now to deal with warnings/errors/exceptions:
- raise an exception
- set self.exit_code
- return warning boolean